lib/cfa/matcher.rb

```diff
@@ -41,4 +75,14 @@ def value_match?(element, matcher)
       end
     end
   end
+
+  # An abstract class documenting the block argument to {Matcher#initialize}
+  class MatcherBinaryPredicate < Proc
```

Comment:
for me cost is confusion for user as it can look that you cannot use common proc or common Hash, which is not true. And when you redefine method, it can make impression we have own specialized implemention of that method.
For documenting block e.g. I prefer to use `@yield`, `@yieldreturn` and `@yieldparam`